components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -174,34 +174,28 @@ The following formula describes the storage update fee for a transaction:
 
 [stem]
 ++++
-\underbrace{\textit{gas_price}(\text{calldata_cost} \cdot n + 200 \cdot n)}_{\text{contract addresses +  new nonce and number of storage updates
+\underbrace{\textit{gas_price}\left(\text{calldata_cost} \cdot 2n - \text{contract_update_discount}\cdot n\right)}_{\text{contract addresses +  new nonce and number of storage updates
 }} \\
 
 + \\
 
-\underbrace{\textit{gas_price} \cdot (\text{calldata_cost}(2m-1)+272)}_{\text{storage updates}}
+\underbrace{\textit{gas_price} \cdot \left(\text{calldata_cost}(2(m-2))+310\right)}_{\text{storage updates}}
 ++++
 
 where:
 
 * stem:[$n$] is the number of unique contracts updated.
-* stem:[$m$] is the number of unique values updated. Be aware that the account balance is always updated, because the account needs to pay a fee. A discount is applied to the account balance update to consider that the value of the account balance most likely does not use the full 32 bytes allotted to it.
-* stem:[$\text{calldata_cost}$] is 512 gas per 32-byte word.
-* stem:[$200$] is the cost of the onchain storage update's meta information and nonce, in gas. This value accounts for the fact that out of 32 bytes in the meta information, only the following six bytes are non-zero:
+* stem:[$m$] is the number of values updated, not counting multiple updates for the same key. Note that stem:[$m\ge 2$] always holds since the sending account balance and the sequencer's balance are always updated. The sender's balance is treated separately (described below) and the sequencer's balance is ignored since it is touched by all the transactions within a block.
```

Comment:
```suggestion
* stem:[$m$] is the number of values updated, not counting multiple updates for the same key. Notice that stem:[$m\ge 2$] is always true,  because the balance of the account sending the transaction and the sequencer's balance are always updated. The sender's balance is treated separately, as described below, and the sequencer's balance is ignored since it is touched by all the transactions within a block.
```